contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol

```diff
@@ -639,4 +680,133 @@ contract ERC20BridgeSampler is
     {
         require(makerToken != takerToken, "ERC20BridgeSampler/INVALID_TOKEN_PAIR");
     }
+
+    function _sampleSellForApproximateBuy(
+        address takerToken,
+        address makerToken,
+        uint256 makerTokenAmount,
+        bytes4 selector,
+        address plpRegistryAddress
+    )
+        private
+        view
+        returns (uint256 takerTokenAmount)
+    {
+        bytes memory callData;
+        uint256[] memory tmpTakerAmounts = new uint256[](1);
+        tmpTakerAmounts[0] = makerTokenAmount;
+        if (selector == this.sampleSellsFromKyberNetwork.selector) {
+            callData = abi.encodeWithSelector(
+                this.sampleSellsFromKyberNetwork.selector,
+                takerToken,
+                makerToken,
+                tmpTakerAmounts
+            );
+        } else {
+            callData = abi.encodeWithSelector(
+                this.sampleSellsFromLiquidityProviderRegistry.selector,
+                plpRegistryAddress,
+                takerToken,
+                makerToken,
+                tmpTakerAmounts
+            );
+        }
+        (bool success, bytes memory resultData) = address(this).staticcall(callData);
+        if (!success) {
+            return 0;
+        }
+        // solhint-disable indent
+        takerTokenAmount = abi.decode(resultData, (uint256[]))[0];
+    }
+
+    function _sampleApproximateBuysFromSource(
+        address takerToken,
+        address makerToken,
+        uint256[] memory makerTokenAmounts,
+        FakeBuyOptions memory opts,
+        bytes4 selector,
+        address plpRegistryAddress
+    )
+        private
+        view
+        returns (uint256[] memory takerTokenAmounts)
+    {
+        _assertValidPair(makerToken, takerToken);
+        if (makerTokenAmounts.length == 0) {
+            return takerTokenAmounts;
+        }
+        uint256 sellAmount;
+        uint256 buyAmount;
+        uint256 slippageFromTarget;
+        takerTokenAmounts = new uint256[](makerTokenAmounts.length);
+        sellAmount = _sampleSellForApproximateBuy(
+            makerToken,
+            takerToken,
+            makerTokenAmounts[0],
```

Comment:
We'd need to be like the Kyber and fetch decimals, and 1 unit of WBTC is A LOT, and it could be greater than the first `makerTokenAmount`, so we'd need to walk back down instead of walking up. 

I'm not sure if I see the gains from 1 unit over working with the first element in `makerTokenAmounts`?

Reply:
This was based on my prev comment, where the sky was falling. Should be good now.